### PR TITLE
[full-ci] [tests-only] Add apisteps user has shared resources stable 6.0

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This PR adds the API step `And "Alice" shares the following resource using API` and its implementation for sharing the resources with the user.

## Related Issue

- Part of https://github.com/owncloud/web/issues/8101

## How Has This Been Tested?

- Locally
- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] E2E tests added

### CODE REVIEW

File: tests/e2e/support/api/share/share.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  shareType,
+  role
+}: {
+  user: User
+  path: string
+  shareWith: string
+  shareType: string
+  role: string
+}): Promise<void> => {
+  const shareTypeToIntString = shareType === 'group' ? '1' : shareType === 'public link' ? '3' : '0'
```
Comment (saw-jan) [inline tests/e2e/support/api/share/share.ts:17]: Best to store shareType object somewhere and use to get the respective type number
```suggestion
// somewhere
export const shareTypes = {
  user: '0',
  group: '1',
  public: '3'
}
```

File: tests/e2e/support/api/share/share.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const body = new URLSearchParams()
+  body.append('path', path)
+  body.append('shareWith', shareWith)
+  body.append('shareType', shareTypeToIntString)
```
Comment (saw-jan) [inline tests/e2e/support/api/share/share.ts:21]: and can be used as
```suggestion
  body.append('shareType', shareTypes[shareType])
```

**Review by saw-jan — COMMENTED** [review]
This would be better

File: tests/e2e/support/api/share/share.ts [hunk 3]
```diff
@@ -1,6 +1,17 @@
...
+  user: string
+  group: string
+  public: string
+  federated: string
+}> = {
+  user: '0',
+  group: '1',
+  public: '3',
+  federated: '6'
+}
```
Comment (saw-jan) [inline tests/e2e/support/api/share/share.ts:14]:
```suggestion
import join from 'join-path'

export const shareTypes: Readonly<{
  user: string
  group: string
  public: string
  federated: string
}> = {
  user: '0',
  group: '1',
  public: '3',
  federated: '6'
}
```

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by ScharfViktor — APPROVED** [review]

**Review by SwikritiT — APPROVED** [review]

**Pull request merged**